code_examples/sdk_examples/src/core_features/messaging/02_encrypt_message.ts

```diff
@@ -0,0 +1,28 @@
+import * as Kilt from '@kiltprotocol/sdk-js'
+import { useEncryptionCallback } from '../signCallback/useEncryptionCallback'
+
+export async function encryptMessage(
+  message: Kilt.IMessage,
+  senderUri: Kilt.DidUri,
+  receiverUri: Kilt.DidUri,
+  keyAgreement: Kilt.KiltEncryptionKeypair
+): Promise<Kilt.IEncryptedMessage> {
+  const { document: senderDocument } = await Kilt.Did.resolve(senderUri)
+
+  const { document: receiverDocument } = await Kilt.Did.resolve(receiverUri)
+
+  const receiverKeyAgreementUri = `${receiverUri}${receiverDocument.keyAgreement?.[0].id}`
+  const senderKeyAgreeementUri = `${senderUri}${senderDocument.keyAgreement?.[0].id}`
+  // encrypt the message
+  const encryptedMessage = await Kilt.Message.encrypt(
+    message,
+    useEncryptionCallback({
+      keyAgreement,
+      keyAgreementUri: senderKeyAgreeementUri as Kilt.DidResourceUri
```

Comment:
Maybe it's better to do the type casting when the variable is created?

Reply:
Yes, you are right.